### ISSUE TITLE
DomainEventReader does not handle "removed" events

### ISSUE THREAD

**Issue by btgoodwin** [RedhawkSDR/redhawk#6]
In Events.cpp, the `DomainEventReader::_addRemoveMsgHandler` method only attempts to unpack an _Added_ event type.  The unpack returns false and skips any _Removed_ event types.

A change like the following will handle both event types:

```
  bool  DomainEventReader::_addRemoveMsgHandler( const CORBA::Any & data ) {
    const StandardEvent::DomainManagementObjectAddedEventType *amsg;
    const StandardEvent::DomainManagementObjectRemovedEventType *rmsg;
    DomainStateEvent evt;
    bool unpacked = false;
    if ( data >>= amsg ) {
      evt.prod_id = amsg->producerId;
      evt.source_id = amsg->sourceId;
      evt.source_name = amsg->sourceName;
      evt.category = amsg->sourceCategory;
      evt.obj = amsg->sourceIOR;
      unpacked = true;
    } else if (data >>= rmsg ) {
      evt.prod_id = rmsg->producerId;      
      evt.source_id = rmsg->sourceId;
      evt.source_name = rmsg->sourceName;
      evt.category = rmsg->sourceCategory;
      evt.obj = CORBA::Object::_nil();
      unpacked = true;
    }

    if (unpacked) {
      AddRemoveListeners::iterator i = addrm_cbs.begin();
      for( ; i != addrm_cbs.end(); i++ ) {
        (*(*i))( evt );
      }
    }
    return unpacked;
  }
```

The issue is present in both 2.0.8 and 2.1.2.

 > EDIT: Added `_nil` for remove events.

**Comment by MichelleD123** [comment]
The REDHAWK team is evaluating the reported issue and submitted patch.

**Comment by MichelleD123** [comment]
My apologies for the error...reopening this ticket.

**Comment by maxrobert** [comment]
The suggested fix resolves the issue and will be applied to a future release

**Comment by shmahon** [comment]
This is being tracked by internal ticket EXT-220 and is expected to be fixed in 2.2.2.

**Comment by shmahon** [comment]
This issue was resolved in release 2.2.3.
